CONTRIBUTING.md

```diff
@@ -325,6 +326,84 @@ if you change the history (rebase, squash, amend), and use git push --force to
 update the branch on your fork. The pull request points to that branch, not to
 specific commits.
 
+## Exercism.io Web Structure
+
+### Intro
+Exercism is built with the [Sinatra](https://github.com/sinatra/sinatra) web framework.
+If you haven't heard of it, definitely check it out. It's an awesome lightweight web frameworks for Ruby.
+
+Exercism seems to follow what some may call an MVP (Model-View-Presenter) architecture.
+Read more about that [here](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93presenter).
+
+### Base Layout (Where the Magic Happens)
+
+#### App (Routes, Presenters and Views)
+App is the user facing side of exercism.
+It handles logging in (and other functions), making calls out to the database to store or get information,
+structuring HTML (that can later be styled), etc.
+
+**Helpers** can be anything that *helps* presentation of data. For exercism, these are things like:
+wrappers for Markdown and Syntax Highlighting parsers, session handling, etc.
+Helpers are located in `app/helpers.rb` If you add a helper, update the list of existing helpers in this file (the convention is `ClassName: 'filename'`).
```

Comment:
Helpers are defined in individual modules under `app/helpers/`. The full list of helpers is in `app/helpers.rb`, and if you add a new helper, this file needs to be edited. The convention is...